domains/client/domain-executor/src/domain_bundle_proposer.rs

```diff
@@ -131,16 +130,16 @@ where
     }
 
     /// Returns the receipts in the next domain bundle.
-    fn collect_bundle_receipts<PBlock, ParentChain, ParentChainHash>(
+    fn collect_bundle_receipts<PBlock, ParentChain, ParentChainBlock>(
         &self,
         header_number: NumberFor<Block>,
         parent_chain: ParentChain,
-        parent_chain_block_hash: ParentChainHash,
+        parent_chain_block_hash: ParentChainBlock::Hash,
```

Comment:
This argument can be removed as the receipts are always expected to be collected against the latest parent chain state, you can use `parent_chain.best_hash()` inside this function. In this way, a few things can be simplified further.

```suggestion
```